src/new-components/password-field/password-field.test.tsx

```diff
@@ -0,0 +1,235 @@
+        render(
+            <>
+                <PasswordField
+                    data-testid="password-field"
+                    label="Phone"
+                    hint="So we can call you"
+                    aria-describedby="custom-hint"
+                />
+                <div id="custom-hint">This is the phone where we will call you</div>
+            </>,
+        )
+        expect(screen.getByTestId('password-field')).toHaveAccessibleDescription(
+            'This is the phone where we will call you',
+        )
+    })
+
+    it('renders its auxiliary label', () => {
+        render(
+            <PasswordField
+                label="New Password"
+                auxiliaryLabel={<a href="/help">Whatʼs this?</a>}
+            />,
+        )
+        expect(screen.getByRole('link', { name: 'Whatʼs this?' })).toBeInTheDocument()
+    })
+
+    it('does not use the auxiliary label for semantic labelling purposes', () => {
+        render(
+            <PasswordField
+                data-testid="password-field"
+                label="New Password"
+                auxiliaryLabel={<a href="/help">Whatʼs this?</a>}
+            />,
+        )
+        expect(screen.getByTestId('password-field')).toHaveAccessibleName('New Password')
+        expect(screen.getByTestId('password-field')).not.toHaveAccessibleDescription()
+    })
+
+    it('can have a placeholder text', () => {
+        render(
+            <PasswordField
+                label="Email"
+                data-testid="password-field"
+                placeholder="Enter your email address"
+            />,
+        )
+        expect(screen.getByTestId('password-field')).toBe(
+            screen.getByPlaceholderText('Enter your email address'),
+        )
+    })
+
+    it('renders an input with type="password" and does not allow providing an alternative type', () => {
+        const { rerender } = render(<PasswordField data-testid="password-field" label="Password" />)
+
+        const passwordField = screen.getByTestId('password-field')
+        expect(passwordField).toHaveAttribute('type', 'password')
+
+        rerender(
+            <PasswordField
+                label="Password"
+                // @ts-expect-error
+                type="text"
+            />,
+        )
+        expect(passwordField).toHaveAttribute('type', 'password')
+    })
+
+    it('allows to toggle visibility of the password value', () => {
+        render(<PasswordField data-testid="password-field" label="Password" />)
+
+        const passwordField = screen.getByTestId('password-field')
+        const togglePasswordButton = screen.getByRole('button', {
+            name: 'Toggle password visibility',
+        })
+
+        expect(passwordField).toHaveAttribute('type', 'password')
+
+        userEvent.click(togglePasswordButton)
+        expect(passwordField).toHaveAttribute('type', 'text')
+
+        userEvent.click(screen.getByRole('button', { name: 'Toggle password visibility' }))
```

Comment:
I think we can either have a separate check for it or just pass a different value to `togglePasswordLabel` property to make sure it renders correct label.

Reply:
Very good point.